src/TextField/TextFieldInput.js

```diff
@@ -121,7 +121,12 @@ export default class TextFieldInput extends Component {
   }
 
   checkDirty(obj) {
-    if (this.props.onDirty && isDirty(obj)) {
+    const dirty = isDirty(obj);
+    if (dirty === undefined) {
```

Comment:
Right, that's an issue with the component in the first place.
The cleaner fix seems to follow the `componentWillMount` hook with:
```jsx
  componentWillUpdate(nextProps) {
    if (this.isControlled()) {
      this.checkDirty(nextProps);
    }
  }
```
You can remove this line.